rosidl_runtime_rs/src/sequence.rs

```diff
@@ -261,18 +261,26 @@ where
     ///
     /// Equivalent to `&seq[..]`.
     pub fn as_slice(&self) -> &[T] {
-        // SAFETY: self.data points to self.size consecutive, initialized elements and
-        // isn't modified externally.
-        unsafe { std::slice::from_raw_parts(self.data, self.size) }
+        if self.data.is_null() {
+            &[]
```

Comment:
Taken from this example in the stdlib

https://doc.rust-lang.org/std/slice/fn.from_raw_parts.html#ffi-handling-null-pointers